App/screens/ProofRequest.tsx

```diff
@@ -19,109 +25,143 @@ interface Props {
+  }
+
+  const transformAttributes = (attributes: Record<string, RequestedAttribute[]>): CredentialDisplay[] => {
+    const transformedAttributes = []
+    for (const attribute in attributes) {
+      transformedAttributes.push({
+        name: attribute,
+        value: attributes[attribute][0].credentialInfo.attributes[attribute],
+        credentialDefinitionId: parseSchema(attributes[attribute][0].credentialInfo.schemaId),
+      })
+    }
+    return transformedAttributes
+  }
+
+  const getProofRecord = (proofId?: string): ProofRecord | void => {
+    try {
+      if (!proofId) {
+        throw new Error(t('ProofRequest.ProofNotFound'))
+      }
+      return useProofById(proofId)
+    } catch (e: unknown) {
       Toast.show({
-        type: 'success',
-        text1: t('ProofRequest.SuccessfullyAcceptedProof'),
+        type: 'error',
+        text1: (e as Error)?.message || t('Global.Failure'),
       })
       navigation.goBack()
     }
-  }, [proof])
+  }
+
+  const getConnectionRecordFromProof = (connectionId?: string): ConnectionRecord | void => {
+    if (connectionId) {
+      return useConnectionById(connectionId)
+    }
+  }
 
-  const getRetrievedCredentials = async () => {
+  const getRetrievedCredentials = async (proof: ProofRecord) => {
     try {
-      if (!proof?.requestMessage?.indyProofRequest) {
-        Toast.show({
-          type: 'error',
-          text1: t('ProofRequest.ProofNotFound'),
-        })
-        throw new Error('Indy proof request not found')
+      const creds = await agent.proofs.getRequestedCredentialsForProofRequest(proof.id)
+      if (!creds) {
+        throw new Error(t('ProofRequest.RequestedCredentialsCouldNotBeFound'))
       }
-      const retrievedCreds = await agent?.proofs?.getRequestedCredentialsForProofRequest(
-        proof?.requestMessage?.indyProofRequest,
-        undefined
-      )
-      if (!retrievedCreds) {
-        Toast.show({
-          type: 'error',
-          text1: t('ProofRequest.RequestedCredsNotFound'),
-        })
-        throw new Error('Retrieved creds not found')
-      }
-      setRetrievedCredentials(retrievedCreds)
-      setRetrievedCredentialsDisplay(transformAttributes(retrievedCreds?.requestedAttributes))
-    } catch {
+      setRetrievedCredentials(creds)
+      setRetrievedCredentialsDisplay(transformAttributes(creds.requestedAttributes))
+    } catch (e: unknown) {
       Toast.show({
         type: 'error',
-        text1: t('Global.Failure'),
+        text1: (e as Error)?.message || t('Global.Failure'),
       })
     }
   }
 
+  const proof = getProofRecord(route?.params?.proofId)
+
+  if (!proof) {
+    Toast.show({
+      type: 'error',
+      text1: t('ProofRequest.ProofNotFound'),
+    })
+    navigation.goBack()
+    return null
+  }
+
   useEffect(() => {
-    getRetrievedCredentials()
+    try {
+      getRetrievedCredentials(proof)
+    } catch (e: unknown) {
+      navigation.goBack()
+    }
   }, [])
 
+  useEffect(() => {
```

Comment:
Curious, why two `useEffect` rather than one with a conditional or switch statement for the different proof states?

Reply:
Tbh, I just found it cleaner to have them isolated (I observed the pattern previously used before I made changes), each effect dealing with a single state.